diff --git a/mconfig b/mconfig
index 2a72625d9..a6d4bab19 100755
--- a/mconfig
+++ b/mconfig
@@ -660,7 +660,6 @@ fi

 if [ "$found_modules" != "" ]; then
        echo " found build modules:"
-       echo "CPPFLAGS :=" >> $makeit_makefile_base
        # NOTE: parsed module (*.mconf) files only substitute var tgt_arch for now
        for m in $found_modules; do
                mod=`eval echo ${m##*./}`
@@ -684,9 +683,7 @@ fi

 # init LDFLAGS if static builds are supported
 if [ "$hststatic" = "1" -a "$tgtstatic" = "1" ]; then
-       echo "LDFLAGS := -static $ldflags" >> $makeit_makefile_base
-else
-       echo "LDFLAGS := $ldflags" >> $makeit_makefile_base
+       echo "LDFLAGS += -static $ldflags" >> $makeit_makefile_base
 fi
 echo >> $makeit_makefile_base

diff --git a/mlocal/frags/common_opts.mk b/mlocal/frags/common_opts.mk
index 64160bf10..352ae07c5 100644
--- a/mlocal/frags/common_opts.mk
+++ b/mlocal/frags/common_opts.mk
@@ -2,6 +2,6 @@
 CLEANFILES :=

 # general build-wide compile options
-AFLAGS := -g
+AFLAGS += -g

 LDFLAGS += -Wl,-z,relro,-z,now
