diff --git a/mlocal/frags/Makefile.stub b/mlocal/frags/Makefile.stub
index f419f030..04e14ba2 100644
--- a/mlocal/frags/Makefile.stub
+++ b/mlocal/frags/Makefile.stub
@@ -89,7 +89,7 @@ cni_config_LIST := $(SOURCEDIR)/etc/network/00_bridge.conflist \
                    $(SOURCEDIR)/etc/network/10_ptp.conflist \
                    $(SOURCEDIR)/etc/network/20_ipvlan.conflist \
                    $(SOURCEDIR)/etc/network/30_macvlan.conflist
-cni_config_INSTALL := $(DESTDIR)$(SYSCONFDIR)/singularity/network
+cni_config_INSTALL := $(DESTDIR)$(SYSCONFDIR)/cni/net.d
 
 actions_LIST := $(SOURCEDIR)/etc/actions/exec $(SOURCEDIR)/etc/actions/run $(SOURCEDIR)/etc/actions/shell \
 	$(SOURCEDIR)/etc/actions/start $(SOURCEDIR)/etc/actions/test
@@ -97,13 +97,13 @@ actions_INSTALL := $(DESTDIR)$(SYSCONFDIR)/singularity/actions
 
 INSTALLFILES := $(singularity_INSTALL) $(starter_INSTALL) $(starter_suid_INSTALL) $(sessiondir) \
 	$(config_INSTALL) $(dist_bin_SCRIPTS_INSTALL) $(capability_JSON) $(syecl_config_INSTALL) \
-	$(bash_completion_INSTALL) $(actions_INSTALL) $(cni_plugins_INSTALL) $(cni_config_INSTALL) \
+	$(bash_completion_INSTALL) $(actions_INSTALL) $(cni_config_INSTALL) \
 	$(seccomp_profile_INSTALL) $(NVIDIA_liblist_INSTALL) $(cgroups_config_INSTALL) 
 
 CLEANFILES += $(libruntime) $(starter) $(singularity) $(go_BIN) $(go_OBJ) $(bash_completion) \
 	$(cni_plugins_EXECUTABLES)
 
-all: cscope collect $(libruntime) $(go_BIN) $(config) $(starter) $(bash_completion) cniplugins
+all: cscope collect $(libruntime) $(go_BIN) $(config) $(starter) $(bash_completion)
 
 .PHONY: cniplugins
 cniplugins:
diff --git a/src/pkg/network/network.go b/src/pkg/network/network.go
index 99638dca..470ef354 100644
--- a/src/pkg/network/network.go
+++ b/src/pkg/network/network.go
@@ -47,10 +47,10 @@ type portMap struct {
 }
 
 // DefaultCNIConfPath is the default directory to CNI network configuration files
-var DefaultCNIConfPath = filepath.Join(buildcfg.SYSCONFDIR, "singularity", "network")
+var DefaultCNIConfPath = filepath.Join(buildcfg.SYSCONFDIR, "cni", "net.d")
 
 // DefaultCNIPluginPath is the default directory to CNI plugins executables
-var DefaultCNIPluginPath = filepath.Join(buildcfg.LIBEXECDIR, "singularity", "cni")
+var DefaultCNIPluginPath = filepath.Join(buildcfg.PREFIX, "lib", "cni")
 
 // AvailableNetworks lists configured networks in configuration path directory
 // provided by cniPath
